### PR TITLE
New git ssh cheat sheet

### PR DESCRIPTION
**Pull request by marmt**
https://duck.co/ia/view/git_ssh_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@marmt Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by altern8tif** [comment]
@marmt the tests are failing because your file name should be `git-ssh.json` instead of `git_ssh.json`.

**Comment by marmt** [comment]
@altern8tif Thanks that did it!

File: share/goodie/cheat_sheets/json/git-ssh.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{
+    "id": "git_ssh_cheat_sheet",
+    "name": "git ssh",
+    "description": "A quick reference to generate ssh keys and use ssh-agent to connect to github.",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/git-ssh.json]: nit: let's remove the period, and capitalize GitHub

File: share/goodie/cheat_sheets/json/git-ssh.json [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    "metadata": {
+        "sourceName": "github.com",
+        "sourceUrl" : "https://help.github.com/articles/generating-an-ssh-key/"
+    },
+
+    "aliases": [
+        "ssh keys", "ssh-keygen", "ssh-agent", "generate ssh keys"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/git-ssh.json]: I think these aliases are more appropriate for the existing SSH Cheat Sheet? https://duckduckgo.com/?q=ssh+cheat+sheet&ia=cheatsheet&iax=1

File: share/goodie/cheat_sheets/json/git-ssh.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            
+        ],
+        "Test the Connection": [
+            {
+                "val": "ssh to github",
+                "key": "ssh -T git@github.com"
+            },
+            {
+                "val": "Success!",
+                "key": "Hi username! You've successfully authenticated, but GitHub does not provide shell access."
```
Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/git-ssh.json]: Quoi? I don't think we want this...?

File: share/goodie/cheat_sheets/json/git-ssh.json (hunk 2 above)
Comment (marmt) [inline share/goodie/cheat_sheets/json/git-ssh.json]: Ok was not sure about that as there are no ssh-keygen commands included.

File: share/goodie/cheat_sheets/json/git-ssh.json (hunk 1 above)
Comment (marmt) [inline share/goodie/cheat_sheets/json/git-ssh.json]: done

File: share/goodie/cheat_sheets/json/git-ssh.json (hunk 3 above)
Comment (marmt) [inline share/goodie/cheat_sheets/json/git-ssh.json]: want what?

Comment (GuiltyDolphin) [inline share/goodie/cheat_sheets/json/git-ssh.json]: The 'Success' section. It seems very... Off.

Comment (marmt) [inline share/goodie/cheat_sheets/json/git-ssh.json]: Yeah not really a terminal command.

File: share/goodie/cheat_sheets/json/git-ssh.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    "id": "git_ssh_cheat_sheet",
+    "name": "git ssh",
+    "description": "A quick reference to generate ssh keys and use ssh-agent to connect to GitHub",
+
+    "metadata": {
+        "sourceName": "github.com",
+        "sourceUrl" : "https://help.github.com/articles/generating-an-ssh-key/"
+    },
+
+    "aliases": [
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/git-ssh.json]: nit: this can be removed if it's empty

File: share/goodie/cheat_sheets/json/git-ssh.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
+{
+    "id": "git_ssh_cheat_sheet",
+    "name": "git ssh",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/git-ssh.json]: nit: I suggest we capitalize this: `Git SSH`

**Comment by moollaza** [comment]
@marmt This LGTM :+1:

Thanks again for taking the time to contribute! We really appreciate it :smiley:

**Comment by marmt** [comment]
I'm thinking about removing the list keys section and get right to the heart of the matter with ssh-keygen. The list keys command is rudimentary and probably unnecessary.

**Comment by marmt** [comment]
@moollaza Thanks! I'm really enjoying the experience and the chance to contribute. It also really helped getting my dev workflow in place.

File: share/goodie/cheat_sheets/json/git-ssh.json (hunk 5 above)
Comment (marmt) [inline share/goodie/cheat_sheets/json/git-ssh.json]: @moollaza What does lgtm +1 mean?

**Pull request merged**